### ISSUE TITLE
Queue understanding

### ISSUE THREAD

**Issue by theobouwman** [jessetane/queue#22]
Why do I have to call `q.start()` when I added a new item to the queue?

**Comment by jessetane** [comment]
For increased flexibility - what if you added jobs to a queue inside a loop and those jobs called their callbacks synchronously?

**Comment by jessetane** [comment]
If you add jobs to a queue that has already been started and hasn't finished yet, you don't have to call start.

**Comment by garygreen** [comment]
@jessetane this has me confused as well. My use case -- I'm trying to generate thumbnails by looping through a directory and want to limit number of thumbs generated at any given time. I can use `readdirSync` but it would be nicer to use `readdir` which means I can add stuff to the queue and it'll chug along as and when the directory is transversed. With the sync version it'll have to wait for the directory structure to be fully resolved before it can even start the queue, this is particularly slow if recursively looking through.

**Comment by jessetane** [comment]
Would something like this work for you?
```javascript
q.auto = function (job) {
  this.push(job)
  this.start()
}
```

**Comment by garygreen** [comment]
I think that could work. Would calling `start` multiple times in the below have any weird consequences?

```js
var queue = queue({
    concurrency: 15
});

readdir('images', function(err, files) {
    if (err) {
        return;
    }
    files.forEach(function(file) {
        queue.auto(function(cb) {
           createThumb(file, cb);
         });
    });
});
```

**Comment by jessetane** [comment]
Not that I can think of.

**Comment by kosssi** [comment]
maybe you can add parameter: 

```
var queue = queue({
    concurrency: 15,
    autostart: true
});
```

and add an event for finish

```
queue.on('finish', function(err) {
    console.log('all task is finished!', err);
});
```

after you can just push task.

```
queue.push(...);
```

What do you think ?

**Comment by jessetane** [comment]
Happy to look at a PR if anyone is up for it

**Comment by tnguyen14** [comment]
My first attempt at a PR #28.

**Comment by jessetane** [comment]
Fixed by #28

**Comment by kosssi** [comment]
👍
